Stardrop/ViewModels/ModDownloadViewModel.cs

```diff
@@ -0,0 +1,162 @@
+﻿using ReactiveUI;
+using System;
+using System.Linq;
+using System.Reactive;
+using System.Reactive.Linq;
+using System.Threading;
+
+namespace Stardrop.ViewModels
+{
+    public enum ModDownloadStatus
+    {
+        NotStarted,
+        InProgress,
+        Successful,
+        Canceled,
+        Failed
+    }
+
+    public class ModDownloadViewModel : ViewModelBase
+    {
+        private readonly DateTimeOffset _startTime;
+        private readonly CancellationTokenSource _downloadCancellationSource;
+
+        // Communicates up to the parent panel that the user wants to remove this download from the list.
+        public event EventHandler? RemovalRequested = null!;
+
+        // --Set-once properties--
+        public Uri ModUri { get; init; }
+
+        public string SuccessStatusText { get; init; } = Program.translation.Get("ui_downloads_panel.download_success");
```

Comment:
One thing that probably needs fixing--this binds to the language on startup, but doesn't update when the user changes language. What's the pattern for making sure my data-bound strings stay up to date?

Reply:
ah, totally missed the `TranslateExtension`. Addressed... most of these. There are one or two that are trickier. I'll call them out inline.